mlir/include/mlir/IR/PDLPatternMatch.h.inc

```diff
@@ -0,0 +1,992 @@
+//===- PDLPatternMatch.h - PDLPatternMatcher classes -------==---*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef MLIR_IR_PDLPATTERNMATCH_H
+#define MLIR_IR_PDLPATTERNMATCH_H
+
+#if MLIR_ENABLE_PDL
+#include "mlir/IR/Builders.h"
+#include "mlir/IR/BuiltinOps.h"
```

Comment:
There is no include from the to the PDL dialect: where is the coupling?

Reply:
These are structures that only make sense with PDL, I want these all to be clearly stubbed out so that it returns empty arrays etc. Could probably leave in and hope DCE does its thing during compilation, but wanted to be more explicit.